### PR TITLE
feat(decisioning): add crash-safe push task settlement

### LINKED ISSUE #2730: decisioning: support crash-safe out-of-process settlement for push-enabled tasks
## Problem

Beta.14 correctly makes polling-only out-of-process settlement use a persisted
`ScopedTaskRef`, but the trusted worker helpers deliberately reject tasks with
`hasWebhook: true`:

> Direct scoped worker settlement is unavailable for tasks with push
> notifications because registry-only writes cannot durably deliver the
> terminal webhook. Return the result through the live handoff instead.

The documented alternative is not restart-safe. The handoff completion promise
is process-local; after the request process restarts, a durable approval/HITL
worker can recover the work and its `ScopedTaskRef`, but it cannot recover the
original live handoff. That leaves no sanctioned crash-safe settlement path for
a push-enabled task that legitimately spans processes.

An adopter can call `registry.complete()` and emit a webhook separately, but
that recreates the exact durability gap the helper is protecting against:

1. The terminal registry write commits.
2. The process dies before the terminal webhook is durably bound/enqueued.
3. The task is terminal on polling, but the buyer never receives the promised
   push notification.

Reversing the order creates the inverse inconsistency. A Redis webhook outbox
and a Postgres task registry cannot share a transaction, so application code
cannot make this safe merely by ordering two calls.

## Requested design

Provide an SDK-owned or explicitly supported adopter integration for durable,
out-of-process settlement of push-enabled tasks. The exact API is open, but it
needs a recoverable state machine rather than two best-effort calls. Possible
shapes include:

- a terminal-delivery intent persisted with the task transition and consumed by
  the webhook recovery worker;
- a Postgres outbox option sharing the registry transaction;
- or an adopter-supplied durable settlement coordinator with an SDK-enforced
  prepare/commit/recovery contract.

The worker-facing result should distinguish both task mutation state
(`applied`, compatible/conflicting `already_terminal`, `not_found_in_scope`)
and terminal push-delivery state (`durably_bound`, `delivered`, `recoverable`).

## Acceptance criteria

- A push-enabled task can be settled by a different process after the request
  process restarts.
- No reported terminal task can permanently lack a recoverable terminal push
  delivery.
- Retries after crashes at every boundary are idempotent and preserve one
  canonical delivery identity/payload fingerprint.
- Conflicting terminal settlements do not emit the wrong terminal webhook.
- Unknown/wrong scoped refs remain non-enumerating and are never retried with a
  modified scope.
- Push authentication material remains protected at rest and has a documented
  retention/rotation lifecycle.
- Real Postgres recovery tests cover crashes before and after task transition,
  outbox binding, publish, and acknowledgement.
- The task-registry scoping guide documents the supported production pattern;
  it should not require a process-local handoff to remain alive for arbitrarily
  long HITL work.

## Why this matters for SDK 14

This blocks replacing application-owned task/webhook settlement bridges with
the beta.14 facade for production approval workflows. Polling-only settlement
is now strong; push-enabled settlement needs the same process-failure guarantee.

### PR DESCRIPTION
**Pull request by bokelley**
Closes #2730

## Summary

- add an external handoff mode that waits for a durable queue commit before returning `submitted`
- atomically settle scoped PostgreSQL tasks and checkpoint their protected webhook recovery outbox
- make retries, conflicts, recovery leases, and legacy operation identity fail closed and restart-safe
- add cross-process PostgreSQL, crash-window, tenant-scope, poison-payload, and secret-adapter regressions
- document the worker lifecycle, migrations, outcomes, and operator error policy

## Validation

- `npm run build`
- `npm run typecheck`
- `npm run ci:docs-check`
- full `npm test` passed before the final fail-closed refinements; focused final suites passed:
  - 29 webhook delivery/recovery tests
  - 3 PostgreSQL settlement edge-case tests
  - 3 external handoff lifecycle tests
- protocol, code, DX, and security expert reviews plus `npm run review:codex -- --all --base main`

### CODE REVIEW

File: src/lib/server/webhook-delivery/recovery.ts [hunk 1]
```diff
@@ -596,9 +700,61 @@ function snapshotFingerprint(snapshot: StoredWebhookDeliverySnapshot): string {
...
+  } catch (cause) {
+    throw new WebhookAuthenticationProtectionError('Webhook authentication protection failed', { cause });
+  }
+  if (!protectedAuth || typeof protectedAuth.fingerprint !== 'string' || protectedAuth.fingerprint.length < 16) {
+    throw new TypeError(
+      'WebhookAuthenticationAdapter.protect() must return a stable fingerprint of at least 16 characters'
+    );
+  }
+  canonicalJsonSha256(protectedAuth.protectedValue);
+  const cleartext = authentication.type === 'bearer' ? authentication.token : authentication.secret;
```
[the remark this answers is not in the document]
Comment (bokelley) [reply src/lib/server/webhook-delivery/recovery.ts:724]: Fixed in bd8e111b9. The generic recovery helper no longer enforces a new credential-length minimum; push settlement continues to validate its protocol-specific token length separately. The durability regression now exercises the original 15-character credential again. Verified with build, typecheck, and all 29 webhook-delivery durability tests.

File: src/lib/server/decisioning/runtime/postgres-task-settlement.ts [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+          status: terminal.status,
+          timestamp: typeof existingTimestamp === 'string' ? existingTimestamp : new Date().toISOString(),
+          protocol: protocolForTool(row.tool),
+          result,
+          ...(push.token !== undefined && { token: push.token }),
+          ...(terminal.status === 'failed' && { message: error!.message }),
+        };
+        let prepared: PreparedWebhookDeliverySnapshot;
+        try {
+          prepared = await recovery.prepare(
```
[the remark this answers is not in the document]
Comment (bokelley) [reply src/lib/server/decisioning/runtime/postgres-task-settlement.ts:241]: Tracked as #2737. The transaction/outbox atomicity in this PR is retained for beta.15; the follow-up will move or bound the adopter-owned KMS call without weakening retry fingerprint or race guarantees.

File: .changeset/calm-tasks-recover.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+'@adcp/sdk': minor
+---
+
+Add crash-safe PostgreSQL settlement for push-enabled decisioning tasks. The
+new coordinator commits terminal task state and the durable webhook outbox in
+one transaction, reports task compatibility separately from delivery state,
+protects task-webhook validation tokens at rest, and withholds the submitted
+response until the external producer's durable queue write succeeds. Recovery
+checkpoints now reject webhook authentication credentials shorter than 16
+characters, including callers of the existing generic checkpoint API.
```
[the remark this answers is not in the document]
Comment (bokelley) [reply .changeset/calm-tasks-recover.md]: Fixed in 72566ef89: the stale credential-minimum sentence was removed, so the changeset now matches the shipped compatibility behavior.

File: src/lib/server/webhook-delivery/recovery.ts [hunk 4]
```diff
@@ -596,9 +700,58 @@ function snapshotFingerprint(snapshot: StoredWebhookDeliverySnapshot): string {
...
+  if (
```
[the remark this answers is not in the document]
Comment (bokelley) [reply src/lib/server/webhook-delivery/recovery.ts:725]: Fixed in bd8e111b9: the generic recovery path again accepts its prior short credentials, while protocol-specific push-token validation remains scoped to push settlement.
